Source/WebCore/platform/ScrollbarsController.h

```diff
@@ -109,9 +110,16 @@ class ScrollbarsController {
 
     WEBCORE_EXPORT virtual void updateScrollbarStyle() { }
 
+    WEBCORE_EXPORT virtual bool isRemoteScrollbarsController() const { return false; }
+
 private:
     ScrollableArea& m_scrollableArea;
     bool m_scrollbarAnimationsUnsuspendedByUserInteraction { true };
 };
 
 } // namespace WebCore
+
+#define SPECIALIZE_TYPE_TRAITS_IS_REMOTE_SCROLLBARS_CONTROLLER(ToValueTypeName, predicate) \
```

Comment:
Here in the base class the macros should not refer to the derived class. So this should be
`SPECIALIZE_TYPE_TRAITS_SCROLLBARS_CONTROLLER`

Reply:
Also consider if you should add a `SPECIALIZE_TYPE_TRAITS_SCROLLBARS_CONTROLLER` to ScrollbarsControllerMac, ScrollbarsControllerMock and any other subclasses, which would allow you to write code like `is<ScrollbarsControllerMock()...`

Reply:
Fixed.